dcos/options.go

```diff
@@ -0,0 +1,68 @@
+package dcos
+
+import (
+	"errors"
+	"os"
+)
+
+// ErrEmptyParam is the error returned if option is used with empty parameter.
+var ErrEmptyParam = errors.New("Error using empty parameter")
+
+// Option is a functional option that configures a Reader.
+type Option func(*dcosUtil) error
+
+// OptionDetectIP sets the location of detectIP script.
+func OptionDetectIP(path string) Option {
+	return func(d *dcosUtil) error {
+		if _, err := os.Stat(path); err != nil {
+			return err
+		}
+
+		d.detectIPLocation = path
+		return nil
+	}
+}
+
+// OptionMasterRoleFile sets a file on a filesystem which describes that a node is a master.
+func OptionMasterRoleFile(path string) Option {
+	return func(d *dcosUtil) error {
+		if path == "" {
```

Comment:
for example: master file may be `/opt/mesosphere/roles/master` and agent file could be `/opt/mesosphere/roles/agent`. If the location for the files changes to `/test/{master,agent}` i want to be able to configure it. But it does not necessary mean that files are in place.